Backend/app/rider-platform/rider-app/Main/src/Beckn/OnDemand/Utils/OnSearch.hs

```diff
@@ -251,11 +251,36 @@ makeLatLong provider vehicleVariant location = do
 
     parseLatLongHelper locId gps providerItems = do
       let providerItem = filter (\item -> maybe False (\locIds -> locId `elem` locIds) item.itemLocationIds) providerItems
-      currVehicleVariant <- maybe (pure Nothing) (fmap Just . getVehicleVariant provider) (listToMaybe providerItem)
-      if currVehicleVariant == Just vehicleVariant
+          vehicleVariants = traverse extractVehicleVariants providerItem
+      if maybe False (\vehVars -> Just vehicleVariant `elem` vehVars) (listToMaybe vehicleVariants)
         then Just <$> Common.parseLatLong gps
         else return Nothing
 
+    extractVehicleVariants item = do
+      fromMaybe
+        []
+        ( item.itemFulfillmentIds
+            >>= ( \itemfullfillment ->
+                    provider.providerFulfillments
+                      >>= ( \provFul ->
+                              Just (filterFulfillmentsByFulfillmentId provFul itemfullfillment)
+                          )
+                )
+        )
+
+    filterFulfillmentsByFulfillmentId providerFulfillments arrFullFIllment = do
+      let result = filter (\fulf -> listToMaybe arrFullFIllment == fulf.fulfillmentId) providerFulfillments
```

Comment:
This will always return single fulfilment, 
because of this check `listToMaybe arrFullFIllment == fulf.fulfillmentId`
so instead of `filter`, you can use `find` here. 
This will also make return type as `Maybe VehicleVariant` and solve your issue of list of lists.